pgxpool/pool_test.go

```diff
@@ -143,6 +143,24 @@ func TestPoolAcquireFuncReturnsFnError(t *testing.T) {
 	require.EqualError(t, err, "some error")
 }
 
+func TestPoolCloseAfterAcquireNilContextPanics(t *testing.T) {
+	t.Parallel()
+
+	pool, err := pgxpool.Connect(context.Background(), os.Getenv("PGX_TEST_DATABASE"))
+	require.NoError(t, err)
+	defer pool.Close()
+
+	c := make(chan struct{}, 1)
+	go func() {
```

Comment:
It doesn't really have to be on a separate goroutine. It can be on the same one. This was just a way to show that there is a resource leak that remains.